### PR TITLE
feat(persistence): make launch_app runnable on tmp directory

### LINKED ISSUE #2843: launch_app using temp directory

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #2843 

Screenshot 2024-04-19 at 11 52 35 AM
Screenshot 2024-04-19 at 11 52 51 AM

By default phoenix uses a temp directory and can be overridden via a flag to `launch_app`

### CODE REVIEW

File: src/phoenix/session/client.py [hunk 1]
```diff
@@ -28,6 +28,7 @@ def __init__(
         self,
         *,
         endpoint: Optional[str] = None,
+        warn_if_not_running: bool = True,
```
Comment (RogerHYang) [inline src/phoenix/session/client.py:31]:
```suggestion
        warn_if_server_not_running: bool = True,
```

**Review by RogerHYang — APPROVED** [review]

File: src/phoenix/server/main.py [hunk 2]
```diff
@@ -126,6 +126,7 @@ def _load_items(
     atexit.register(_remove_pid_file)
 
     parser = ArgumentParser()
+    parser.add_argument("--database", required=False)
```
Comment (axiomofjoy) [inline src/phoenix/server/main.py:129]: I think a more verbose flag would be helpful, e.g., `--database-url` or `--database-connection-url`.

File: src/phoenix/services.py [hunk 3]
```diff
@@ -107,6 +107,7 @@ class AppService(Service):
 
     def __init__(
         self,
+        database: str,
```
Comment (axiomofjoy) [inline src/phoenix/services.py:110]: I think a more verbose variable name is helpful so it's clear it's a string.

**Review by axiomofjoy — APPROVED** [review]

File: src/phoenix/server/main.py (hunk 2 above)
Comment (mikeldking) [reply src/phoenix/server/main.py:129]: I'll do database-url. I struggled with how complicated I wanted this to be but true, It's more clear that it's a string

**Pull request merged**